### PR TITLE
Run gRPC web proxy as part of server process

### PR DESCRIPTION
**Pull request by asutula**
This will allow connections from javascript gRPC clients

### CODE REVIEW

File: api/server/server.go [hunk 1]
```diff
@@ -59,18 +62,21 @@ type Server struct {
...
-	LotusAddress    ma.Multiaddr
-	LotusAuthToken  string
-	GrpcHostNetwork string
-	GrpcHostAddress string
-	RepoPath        string
+	LotusAddress        ma.Multiaddr
+	LotusAuthToken      string
+	GrpcHostNetwork     string
+	GrpcHostAddress     string
+	GrpcServerOpts      []grpc.ServerOption
```
Comment (asutula) [inline api/server/server.go]: Bubbling these config options all the way up so was can easily add tls configuration at some point

File: api/server/server.go [hunk 2]
```diff
@@ -59,18 +62,21 @@ type Server struct {
...
+	GrpcWebProxyAddress string
```
Comment (asutula) [inline api/server/server.go]: Address to run the grpc web proxy server on

File: api/server/server.go [hunk 3]
```diff
@@ -125,6 +131,30 @@ func NewServer(conf Config) (*Server, error) {
...
+		if wrappedServer.IsGrpcWebRequest(r) ||
+			wrappedServer.IsAcceptableGrpcCorsRequest(r) ||
+			wrappedServer.IsGrpcWebSocketRequest(r) {
+			wrappedServer.ServeHTTP(w, r)
+		}
+	})
+	grpcWebProxy := &http.Server{
+		Addr:    conf.GrpcWebProxyAddress,
+		Handler: handler,
+	}
```
Comment (asutula) [inline api/server/server.go]: Wrap the grpc service in the web proxy per https://github.com/improbable-eng/grpc-web/tree/master/go/grpcweb and consistent with how we do it in `go-threads`

File: exe/server/main.go [hunk 4]
```diff
@@ -7,15 +7,17 @@ import (
...
 	logging "github.com/ipfs/go-log"
 	"github.com/textileio/fil-tools/api/server"
 	"github.com/textileio/fil-tools/tests"
-	_ "net/http/pprof"
 )
 
 var (
-	grpcHostAddr = "127.0.0.1:5002"
+	grpcHostAddr     = "0.0.0.0:5002"
+	grpcWebProxyAddr = "0.0.0.0:6002"
```
Comment (asutula) [inline exe/server/main.go]: Updated to `0.0.0.0` so we can accept connection over any network interface.

Comment (asutula) [reply exe/server/main.go]: Also @jsign when this gets merged and deployed to azure, can you open up port 6002?

Comment (jsign) [reply exe/server/main.go]: Sure, will do.

**Pull request merged**